src/Microsoft.Experimental.Collections/Microsoft/Collections/Extensions/OrderedDictionary.cs

```diff
@@ -0,0 +1,514 @@
+            {
+                Entry entry = entries[i];
+                array[i + arrayIndex] = new KeyValuePair<TKey, TValue>(entry.Key, entry.Value);
+            }
+        }
+
+        bool ICollection<KeyValuePair<TKey, TValue>>.Remove(KeyValuePair<TKey, TValue> item)
+        {
+            int index = IndexOf(item.Key);
+            if (index >= 0 && EqualityComparer<TValue>.Default.Equals(_entries[index].Value, item.Value))
+            {
+                RemoveAt(index);
+                return true;
+            }
+            return false;
+        }
+
+        int IList<KeyValuePair<TKey, TValue>>.IndexOf(KeyValuePair<TKey, TValue> item)
+        {
+            int index = IndexOf(item.Key);
+            if (index >= 0 && !EqualityComparer<TValue>.Default.Equals(_entries[index].Value, item.Value))
+            {
+                index = -1;
+            }
+            return index;
+        }
+
+        void IList<KeyValuePair<TKey, TValue>>.Insert(int index, KeyValuePair<TKey, TValue> item) => Insert(index, item.Key, item.Value);
+        #endregion
+
+        private void Resize(int newSize)
+        {
+            int[] newBuckets = new int[newSize];
+            Entry[] newEntries = new Entry[newSize];
+
+            int count = Count;
+            Array.Copy(_entries, newEntries, count);
+            for (int i = 0; i < count; ++i)
+            {
+                ref Entry entry = ref newEntries[i];
+                ref int next = ref newBuckets[entry.HashCode % newSize];
+                entry.Next = next - 1;
+                next = i + 1;
+            }
+
+            _buckets = newBuckets;
+            _entries = newEntries;
+        }
+
+        private int IndexOf(TKey key, out int hashCode)
+        {
+            if (key == null)
+            {
+                throw new ArgumentNullException(nameof(key));
+            }
+
+            IEqualityComparer<TKey> comparer = _comparer;
+            hashCode = (comparer?.GetHashCode(key) ?? key.GetHashCode()) & 0x7FFFFFFF;
+            int[] buckets = _buckets;
+            int index = -1;
+            if (buckets.Length > 0)
+            {
+                int bucket = hashCode % buckets.Length;
+                index = buckets[bucket] - 1;
+                comparer = comparer ?? EqualityComparer<TKey>.Default;
+                Entry[] entries = _entries;
+                while (index >= 0)
+                {
+                    Entry entry = entries[index];
+                    if (entry.HashCode == hashCode && comparer.Equals(entry.Key, key))
+                    {
+                        break;
+                    }
+                    index = entry.Next;
+                }
+            }
+            return index;
+        }
+
+        private bool TryInsert(int? index, TKey key, TValue value, InsertionBehavior behavior)
+        {
+            int i = IndexOf(key, out int hashCode);
+            Entry[] entries = _entries;
+            ++_version;
```

Comment:
You changed all postfix to prefix - personal style? I don't think it matters, although it increases the diff with Dictionary.

Reply:
Just personal style from learning c++ in my college days. I can change it if need be.

Reply:
I think it makes sense to be consistent with Dictionary, specially if we're not fixing the other nits because they came from corefx itself. Like the boundary checks by using uint.

Reply:
Ahh github showed me an old diff. I saw you indeed fixed them :smile: thanks!

Reply:
Our general pattern (in corefx anyway) seems to be to avoid both pre and post fix within an expression (eg `foo = bar++` or `foo = ++bar`) therefore the difference does not matter and we generally pick postfix as a convention.

Of course if the code is just clearer then exceptions are fine.